Prepend key modifier enums with 'SDL_'

Kontrabant opened the pull request:
Prepend the `SDL_keymod` enums with `SDL_` for consistency.

slouken commented:
Can you redo this PR with build-scripts/rename_api.py so SDL_oldnames.h is updated correctly?

Please let me know if you run into any issues with this!

Kontrabant commented:
I rolled back, rebased and did it via the rename_api script. The script worked perfectly with the exception of missing one file: it didn't replace the occurrences in the iOS keyboard demo in Xcode-iOS/Demos/src/keyboard.c (I fixed it manually).

slouken commented:
Ah, great. I’ll add that to the directories to update.

Thanks!
